packages/babel-plugin/src/utils/build-styled-component.ts

```diff
@@ -156,56 +209,6 @@ const handleMemberExpressionInStyledInterpolation = (path: NodePath<t.MemberExpr
   return propsToDestructure;
 };
 
-/**
```

Comment:
As discussed with @MonicaOlejniczak, we're removing this logic as  it replaces any use of destructure aliasing with the original prop var name. It's intention was to reduce output size by preventing something like:
```
{
  name: nickName1,
  name: nickName2,
} = props;
```

However, this sort of occurrence would be quite low in a styled component But also removing aliases could cause collisions with outer scope vars, for example:

```
const width = 10;
const Comp = styled.div`
  width: $({ width }) => ${width + width}px;
`;
```
Without aliasing here things won't work as intended.

Reply:
Just to clarify, the second example you gave does work. Usually lint rules that ban shadowed variables are in place to improve code readability.

Regardless, this optimisation is not needed in compiled and it is generally safer for us to preserve as much user code as possible. Bundlers can optimise the final result as well.

Reply:
I was trying say we can't use both `width`s within the arrow function as without aliasing we can only use the one in the function scope :)